Feature - 1066 Service Actions dialogs

aldipower opened the pull request:
This implements the dialogs/confirm-boxes for the service actions (scale, suspend, destroy).
There is no real action behind them, this will follow next.

Destroy:
![confirm-destroy](https://cloud.githubusercontent.com/assets/859154/15862844/367131f0-2cd1-11e6-8754-61aa4b7c7253.png)

Scale:
![confirm-scale](https://cloud.githubusercontent.com/assets/859154/15862846/3685abda-2cd1-11e6-8940-8f52c2f9595d.png)

Suspend:
![confirm-suspend](https://cloud.githubusercontent.com/assets/859154/15862845/3676d57e-2cd1-11e6-97cc-e0875f286ae0.png)

mesosphere-web commented:
Reviewer: @orlandohohmeier

pierlo-upitup commented:
Shouldn't the heading color always match the severity level? @leemunroe

pierlo-upitup commented:
retest this please

aldipower commented:
@pierlo-upitup @leemunroe Just FYI - Unfortunately the color of `text-warning` (ugly yellow) compared to `button-warning` differs a lot in Canvas UI. So we don't have matching colors in the moment.

pierlo-upitup commented on src/js/components/ServiceDetail.js:
```diff
@@ -8,11 +9,17 @@ import ServiceDetailConfigurationTab from './ServiceDetailConfigurationTab';
+  'onAcceptSuspendConfirmDialog',
+  'onCancelDestroyConfirmDialog',
+  'onCancelSuspendConfirmDialog',
+  'onCloseScaleFormModal',
```
shouldnt this be `onCancelScaleFormModal`?

aldipower replied:
I think the wording for closing a _Modal_ is `close`, we use it like that also at other places.
On the other hand the wording in the _Confirm_ is `cancel`.
So if I change this, the things will be mixed up at another place.
I would like to stick with this.

pierlo-upitup commented on src/js/components/ServiceDetail.js:
```diff
@@ -30,7 +37,10 @@ class ServiceDetail extends mixin(InternalStorageMixin, TabsMixin) {
+      isServiceFormModalShown: false,
+      isServiceDestroyConfirmShown: false,
+      isServiceScaleFormModalShown: false,
+      isServiceSuspendConfirmShown: false
```
😱

aldipower replied:
Yes, really bad.

orlandohohmeier replied:
There is always only one modal open, right? How about one variable holding a modal id or nothing to toggle all these modal states?

aldipower replied:
Good idea, at least this looks much better then. Underlaying problem will remain the same..
Will do it.

pierlo-upitup commented:
Doesn't this PR only implement `ServiceScaleFormModal` ?

leemunroe commented:
Very nice.

* I'm a bit concerned about the color of the scale and suspend buttons, I would expect the scale button the be standard modal call to action color (green)
* Can we change the width of the instance input? Likely never going to be more than 3 characters so I would expect the width to better reflect that

aldipower commented:
@pierlo-upitup No, the other dialogs are `Confirm`-elements from reactjs-components as you can see in the code.

@leemunroe Yeah, I will make the buttons to have the standard color and change the width of the input.

aldipower commented:
@orlandohohmeier I've refactored the show states. Feels much better now.

@leemunroe I've changed the button colors to the `primary`-color as used application wide.
Reducing the width of the input field looks silly though. If you like you could show me what you have exactly in mind?

![confirm-scale](https://cloud.githubusercontent.com/assets/859154/15925726/9e913a1a-2e38-11e6-8130-6b4bd1973222.png)

Notice the funny text.. ;)
![confirm-suspend](https://cloud.githubusercontent.com/assets/859154/15925727/9e92f288-2e38-11e6-8ce4-76f78c22d499.png)

orlandohohmeier commented:
L:eyes:ks good to me! Let's :shipit:

Poltergeist commented:
![Merge](http://i.giphy.com/l2JdSlA1a1zKVAyze.gif)

kennyt commented:
did this PR get merged in with failing integration tests?

Poltergeist commented:
@kennyt This was my bad, there was one test failing in the integration tests. In the last couple of days the integration tests were failing a lot, not because they were actually failing but because they were flaky that was why I probably didn't paid enough attention to them. I am sorry for this and will pay more attention next time.
